userscript/src/components/edit-panel/lg-auto-turn-arrow/hooks/use-lane-guidance-control.ts

```diff
@@ -0,0 +1,54 @@
+import { useMutationObserver } from '@/hooks';
+import {
+  findLaneGuidanceContainerByDirection,
+  findLaneGuidanceControlModelByLanesControl,
+  findLaneGuidanceEditRegionContainer,
+  isLanesControlElement,
+} from '../utils';
+import { useMemo, useState } from 'react';
+import { isElementNode } from '@/utils/dom-node';
+
+function findLanesControlNodeInList(nodeList: NodeList): Element {
+  return Array.prototype.find.call(
+    nodeList,
+    (node: Node) => isElementNode(node) && isLanesControlElement(node),
+  );
+}
+
+export function useLaneGuidanceControlElement(
+  lanesDirection: 'fwd' | 'rev',
+): Element {
+  const laneGuidanceContainer =
+    findLaneGuidanceContainerByDirection(lanesDirection);
+  const editRegionContainer = findLaneGuidanceEditRegionContainer(
+    laneGuidanceContainer,
+  );
+
+  const [lanesControlElement, setLanesControlElement] = useState<Element>(null);
+
+  useMutationObserver(
+    editRegionContainer,
+    ([mutation]) => {
+      const addedNode = findLanesControlNodeInList(mutation.addedNodes);
+      const removedNode = findLanesControlNodeInList(mutation.removedNodes);
+
+      const isAdded = addedNode && !removedNode;
+      const isRemoved = !addedNode && removedNode;
+      const isChanged = isAdded || isRemoved;
+      if (!isChanged) return;
+      setLanesControlElement(isAdded ? addedNode : null);
+    },
+    { childList: true },
+  );
+
+  return lanesControlElement;
```

Comment:
In case `editRegionContainer` will be null, the `useMutationObserver` will be capable of handling that because of a special logic in that hook, which will observe nothing if the target element provided is null.